### ISSUE TITLE
Missing core from xtensa-esp32-none-elf

### ISSUE THREAD

**Issue by IsCoffeeTho** [esp-rs/espup#444]
## Bug description

Upon a fresh install from cargo (`cargo install espup`): `xtensa-esp32-none-elf` has no `core` crate
As well as attempting to remedy the 'may not be installed' message: rust errors saying `toolchain 'esp' does not support components`

## To Reproduce

1. Remove previous installation of espup (if applicable)
2. Remove `~/.rustup/toolchains/esp` (if applicable)
3. `$ cargo install espup` <- `. ~/export-esp.sh` implied
4. Navigate to a clean embedded project
5. `$ cargo +esp build`

## Expected behavior

Clean build of project

## Screenshots

![error[E0463]: can't find crate for `core`](https://github.com/user-attachments/assets/ed110a5c-84de-4e86-974b-d109dd6c0982)

![error: toolchain 'esp' does not support componenents](https://github.com/user-attachments/assets/ba61931c-12f0-4dca-a304-fe784d498a93)

## Environment

- OS: Arch Linux x86_64 (Linux 6.10.5-arch1-1)
- espup version: 0.12.2

**Comment by IsCoffeeTho** [comment]
The previous comment has told me nothing about continuing,

- Did this reproduction consider 'Operating System' and 'Kernel'?
- What are some more steps to continue with resolving this issue?
- Is there more information needed, or am i just stuck in the mud?

Any information and/or questions more than "it works on my machine" would be appreciated

**Comment by SergioGasquez** [comment]
> Did this reproduction consider 'Operating System' and 'Kernel'?

In the dev container uses [`bookworm-slim`](https://hub.docker.com/layers/library/debian/bookworm-slim/images/sha256-70d4c04302bdcd71c4fa21b6c12e99888380a07f04e3d44452b961bca046489d?context=explore)

> What are some more steps to continue with resolving this issue?

Have you tried doing `espup install` and sourcing the `export-esp.sh` file? Your reproduction steps do not include this steps.

> Is there more information needed, or am i just stuck in the mud?

I think you provided all the required information, but there is something wrong in the steps. FYI, I use `6.10.6-arch1-1` and `espup 0.12.2` on my local machine and no issues encountered.

**Comment by SergioGasquez** [comment]
> I can compile project on windows but i can't list targets.

Try using `rustc +esp --print target-list `

**Comment by SergioGasquez** [comment]
Closing due to inactivity, if the issue persists feel free to reopen the issue!

**Comment by discapes** [comment]
i'm also getting similar issues
```sh
$ rustup show
Default host: aarch64-unknown-linux-gnu
rustup home:  /home/me/.rustup

installed toolchains
--------------------
stable-aarch64-unknown-linux-gnu (default)
esp (active)

active toolchain
----------------
name: esp
active because: overridden by '/home/me/.../rust-toolchain.toml'
installed targets:
$ rustup target list
error: toolchain 'esp' does not support components
$ rustc --version
rustc 1.92.0 (ded5c06cf 2025-12-08)
$ rustc +esp --print target-list 
error: couldn't read `+esp`: No such file or directory (os error 2)

error: aborting due to 1 previous error
```

**Comment by SergioGasquez** [comment]
> i'm also getting similar issues


Have you installed your toolchain? See https://docs.espressif.com/projects/rust/book/getting-started/toolchain.html#xtensa-devices

**Comment by discapes** [comment]
Yes, multiple times, and that's what was crazy about it. Turns out the issue was `.local/share/cargo/bin/` being in the path *after* `/usr/bin`, so the rustup toolchain wasn't being used even though it was activated in rustup. I thought it wouldn't matter since export-env.sh sets the path:
```sh
export LIBCLANG_PATH="/home/me/.rustup/toolchains/esp/xtensa-esp32-elf-clang/esp-20.1.1_20250829/esp-clang/lib"
export PATH="/home/me/.rustup/toolchains/esp/xtensa-esp-elf/esp-15.2.0_20250920/xtensa-esp-elf/bin:$PATH"
```
But it did, and I fixed it, but still rustup should probably warn about that in commands like rustup show. And the `~/.local/share/cargo/env` tool doesn't do anything if it's in the path in the wrong place.
